.github/workflows/linting.yml

```diff
@@ -116,6 +116,24 @@ jobs:
       - name: Run packagedoc-lint
         run: make packagedoc-lint
 
+  variant-analysis:
+    name: Variant Analysis
```

Comment:
Wouldn’t it be better to give this job a self-explanatory name? As it stands, it only indicates that it’s a variant of some other job. Wouldn’t it be better to name it “Vulnerability Scanning (CodeQL)” (and rename the Anchore job accordingly)?

Reply:
"Variant" in this case isn't meant as "variant of some other job", but instead as the name of a category of static analysis.

> Variant analysis is the process of using a known security vulnerability as a seed to find similar problems in your code. It’s a technique that security engineers use to identify potential vulnerabilities, and ensure these threats are properly fixed across multiple codebases.
Querying code using CodeQL is the most efficient way to perform variant analysis. You can use the standard CodeQL queries to identify seed vulnerabilities, or find new vulnerabilities by writing your own custom CodeQL queries. Then, develop or iterate over the query to automatically find logical variants of the same bug that could be missed using traditional manual techniques.

https://codeql.github.com/docs/codeql-overview/about-codeql/

CodeQL doesn't only do variant analysis for security issues, it also has semantic queries/rules for other types of issues.

https://github.com/github/codeql/tree/main/go/ql/src

I dropped "vulnerability" from the description because I don't think any of the [issues it found in our codebase](https://github.com/submariner-io/submariner/issues/1970#issue-1343160083) were security-related.

I can add some more docs like this to the commit messages.

Reply:
Ah right, TIL, thanks! I hadn’t found anything relevant in a web search, so it might be worth at least linking to the “about CodeQL” page in the commit message.

Reply:
Yeah there aren't a lot of docs, I'll spread some more info around.

Reply:
Added to the commit message :heavy_check_mark: